src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -1407,12 +1409,15 @@ private val RsSelfParameter.canonicalDecl: String
         append("self")
     }
 
-// BACKCOMPAT: ???
-// Fix for IntelliJ platform bug: https://youtrack.jetbrains.com/issue/IDEA-186991
-// replace it with `escapeString()` after the end of support IDEs with the bug
-private fun escapeTy(str: String): String = str
-    .replace("<", "&#60;")
-    .replace(">", "&#62;")
-    .replace("&", "&amp;")
+// BACKCOMPAT: 2020.1. Replace with `escapeString(str)`
+private fun escapeTy(str: String): String {
+    return if (ApplicationInfo.getInstance().build > BUILD_202) {
+        escapeString(str)
+    } else {
+        str.replace("<", "&#60;")
```

Comment:
Oh, I didn't notice this PR last time. In fact, the problem is not that `escapeString(str)` needs to be added, but the original code replacement order is wrong.

```kotlin
str.replace("<", "&#60;").replace(">", "&#62;").replace("&", "&amp;")
```

Such a replacement order will cause & to be escaped incorrectly.

`<` -> `&#60;` -> `&amp;#60;`

Reply:
> In fact, the problem is not that escapeString(str) needs to be added, but the original code replacement order is wrong.

But it's not true. If you just change only order of `replace` calls, it'll fix only #5598 but not #2473 and #5241

![image](https://user-images.githubusercontent.com/2539310/85931370-afc53e80-b8cc-11ea-8f08-1fb926b35fb1.png)